src/java.desktop/macosx/classes/sun/awt/CGraphicsDevice.java

```diff
@@ -306,14 +308,47 @@ public boolean isDisplayChangeSupported() {
         return true;
     }
 
+    /* If the modes are the same or the only difference is that
+     * the new mode will match any refresh rate, no need to change.
+     */
+    private boolean isSameMode(final DisplayMode newMode,
+                               final DisplayMode oldMode) {
+
+        return (Objects.equals(newMode, oldMode) ||
+                (newMode.getRefreshRate() == DisplayMode.REFRESH_RATE_UNKNOWN &&
+                 newMode.getWidth() == oldMode.getWidth() &&
+                 newMode.getHeight() == oldMode.getHeight() &&
+                 newMode.getBitDepth() == oldMode.getBitDepth()));
+    }
+
     @Override
     public void setDisplayMode(final DisplayMode dm) {
         if (dm == null) {
             throw new IllegalArgumentException("Invalid display mode");
         }
-        if (!Objects.equals(dm, getDisplayMode())) {
-            nativeSetDisplayMode(displayID, dm.getWidth(), dm.getHeight(),
-                                 dm.getBitDepth(), dm.getRefreshRate());
+        if (!isSameMode(dm, getDisplayMode())) {
+            try {
+                nativeSetDisplayMode(displayID, dm.getWidth(), dm.getHeight(),
+                                    dm.getBitDepth(), dm.getRefreshRate());
+            } catch (Throwable t) {
+                /* In some cases macOS doesn't report the initial mode
+                 * in the list of supported modes.
+                 * If trying to reset to that mode causes an exception
+                 * try one more time to reset using a different API.
+                 * This does not fix everything, such as it doesn't make
+                 * that mode reported and it restores all devices, but
+                 * this seems a better compromise than failing to restore
+                 */
```

Comment:
Would like to highlight that this tradeoff will break the spec, since we successfully restore the mode which is not in the list of modes.

Reply:
So .. it was in the lsit of modes that the app was handed, and it is a bug that it isn't still there.
But what I've just done is fixed getDIsplayModes() to  include it